cmd/edge-proxy/main.go

```diff
@@ -79,6 +82,17 @@ func main() {
 		os.Exit(1)
 	}
 
+	if externalHTTPProxyURI != "" {
+		parsedExternalProxy, err := url.Parse(externalHTTPProxyURI)
+		_ = parsedExternalProxy //reuslt unused, only concerned with error
+		if err != nil {
+			fmt.Printf("extern-http-proxy-uri invalid: %s\n", err.Error())
+
+			os.Exit(1)
+		}
+		fog_http.SetExternalHTTPProxy(externalHTTPProxyURI)
```

Comment:
Rather than setting a global variable that is picked up by the `EdgeTransport` function, add a new argument to the `EdgeTransport` function and snake the parsed proxy URL to where that is called as with the other configuration.

Reply:
will do.  makes sense